### PR TITLE
Fail fast without create port at all

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

fail fast, inspired by recent discussion in port leak 
for those kind of error handling, we can fail fast without go to create port then delete port stage
if we know it's definitely will fail...

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: pkg/cloud/services/compute/instance.go [hunk 1]
```diff
@@ -193,6 +193,20 @@ func (s *Service) createInstance(eventObject runtime.Object, clusterName string,
...
 	accessIPv4 := ""
 	portList := []servers.Network{}
+	
+	if instanceSpec.Subnet != "" && accessIPv4 == "" {
+		return nil, fmt.Errorf("no ports with fixed IPs found on Subnet %q", instanceSpec.Subnet)
+	}
+
+	imageID, err := s.getImageID(instanceSpec.Image)
+	if err != nil {
+		return nil, fmt.Errorf("create new server: %v", err)
```
Comment (chrischdi) [inline pkg/cloud/services/compute/instance.go:203]: I think that's the wrong message 🙂

**Review by chrischdi — COMMENTED** [review]
Not introduced by your PR but could be easily fixed.

Also ci looks like it wants some gofmt.

File: pkg/cloud/services/compute/instance.go (hunk 1 above)
Comment (jichenjc) [reply pkg/cloud/services/compute/instance.go]: updated :)

**Pull request merged**